git/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+extern crate git2;
+
+use git2::{Error, Repository};
+
+/// Given an existing git repository, it will read the blob that the reference and the filename
+/// point to and return it as a String.
+pub fn cat_file(repo: &Repository, reference: &str, filename: &str) -> Result<String, Error> {
+    let reference = repo.find_reference(reference)?;
+
+    let tree = reference.peel_to_tree()?;
+
+    let path = std::path::Path::new(filename);
+    let te = tree.get_path(path)?;
+    let blob = repo.find_blob(te.id())?;
+
+    Ok(String::from_utf8_lossy(blob.content()).to_string())
```

Comment:
Bit surprised you don't return a `[u8]` from this method as there's every chance this item could be binary, not string (what if I want to read in an asset?)

Maybe then provide a convenience wrapper that returns it as a string, but don't hide the primitive value here?

Reply:
Thinking about it more as well, dunno if we should be using `from_utf8_lossy` or returning an error? I don't think we should _support_ anything other than UTF-8 to start with, but to swallow the errors is probably not right either. Specifically, `from_utf8_lossy` replaces invalid codepoints with `U+FFFD` (the unicode replacement char), which may or may not be safe:

https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_lossy

Reply:
Yup, you are right.
This was a dirty hack I did during the PoC that should be changed. I guess now is as good time as any to change it :-)